src/display/dom_utils.js

```diff
@@ -33,6 +33,37 @@ var createValidAbsoluteUrl = sharedUtil.createValidAbsoluteUrl;
 
 var DEFAULT_LINK_REL = 'noopener noreferrer nofollow';
 
+function DOMCanvasFactory() {}
+DOMCanvasFactory.prototype = {
+  create: function DOMCanvasFactory_create(width, height) {
+    var canvas = document.createElement('canvas');
+    canvas.width = width || 0;
+    canvas.height = height || 0;
```

Comment:
Letting the `width`/`height` breaks `get isFontSubpixelAAEnabled()` case, with *lots* of test failures as a result. I think that you need to do e.g. this instead:
```javascript
if (typeof width !== 'undefined') {
  canvas.width = width;
}
if (typeof height !== 'undefined') {
  canvas.height = height;
}
```